### DISCUSSION TITLE
installing bacula-web on CentOS without a virtual host configuration

### DISCUSSION THREAD

**Discussion by Mehrdad-R10** [bacula-web/bacula-web#221] [General]
Hi
I am trying to set up a test installation of bcula-web on RockyLinux(CentOS/RHEL 9) and not having much luck so far
Problem seems to be my weak grasp of configuring virtual hosting, so I was wondering if it is possible to get bacula-web working without virtual hosting, I know it is preferable from a security point of view but i am not worried about it at this stage, i just need bacula-web to be up and running
Btw, i installed using the composer installation alternative, apache is up and running, but when i get to the bacula-web directory, i just get a directory listing
Appareciate any advice since i seem to be stuck

**Comment by dfranco** [comment]
> Hi
> 
> I am trying to set up a test installation of bcula-web on RockyLinux(CentOS/RHEL 9) and not having much luck so far
> 
> Problem seems to be my weak grasp of configuring virtual hosting, so I was wondering if it is possible to get bacula-web working without virtual hosting, I know it is preferable from a security point of view but i am not worried about it at this stage, i just need bacula-web to be up and running
> 
> Btw, i installed using the composer installation alternative, apache is up and running, but when i get to the bacula-web directory, i just get a directory listing
> 
> Appareciate any advice since i seem to be stuck
> 
> 
> 
> 

Hi @Mehrdad-R10 ,

I would be more than happy to help you.

I know that current documentation needs some more content and fixes (I'm continuously working on this when I have some spare time), but for sure,this is something easy to setup.

Few questions:

- Could you explain how you've setup Apache and PHP ?
- Which version of bacula-web are you using ?

I'll add the missing instructions later in this thread once I have a moment this evening.

**Comment by Mehrdad-R10** [comment]
Hi
Thanks for the reply, didn't expect to get one so fast
As to my problems, i have set up the server according to the instructions,
Rockylinux9/RHEL 9), installed Bacula and mySql which are running,
installed composer, and the rst according to documentation, virtual host
configuration according to documentation up to running a2ensite which is a
debian only thing from what i have gathered,
the tests mentioned in the documentation run and give ok on all points
-User creation works ($ sudo -u apache php bwc setupauth)
-testing set up with (sudo -u www-data php bwc check) gives ok on all points
 but the system does not run
BTW I did manage to run it somewhat now, but it comes to a page with an
error  complaining that a path( i think it is
...../bacula-web/application/assets/protected can not be written to, while
the test above says the path is writable, the whole bacula-web folder is
chown by apache user, almost all catalogs are set to writable, i know it is
not best practices but just trying to see if i can get it to run, but so
far, no success)
PHP is version 8.0.36-something
The installation is new and uptodate on a HyperV VM, given 2GB ram if that
matter, I am aware that it is kinda little for this set up, i may
actualy wigin with a fresh installation and go through installation steps
again, i may have done something wrong, but any advice/insight would be
appreciated

Also, thanks for all the work u have put into this

Sincerely /Mehrdad

**Reply by dfranco** [reply]
Hi @Mehrdad-R10 

Thanks for the kind words about the project.

Regarding the assets/protected error, please make sure you don't have SELinux enforced.

Also, did you specify the basepath in config.php ?

A dump of the web server config and application/config/config.php could be helpful to identify the problem.

Best,

**Comment by Mehrdad-R10** [comment]
Hi David
So, i decided to start again with a fresh vm, one positive side of running
VMs for this kinda work, and this time it worked, not sure if i had missed
something before or what it was but i did two things differently
First, after installing PHP and Apache, i made sure that php and apache
worked, created a simple phpinfo() page and made it work(it did not at
first, not sure why, could have been some mistake i made at installation
but i had to add a few lines to the httpd's conf files)
Second i just added the virtual site as mentioned in in the manuals and
left the debian part of linking the files alone(in my previous installation
i tried to do the equivalent of the debian a2en* and created the sites
enable folders etc., this time, i did not)
Also, i followed the installation steps more carefully  and it worked .
Whether it was for the two things i mentioned above or just following the
instruction more closely, i don't know butit works
I really like the esthetics and page design, it is way cleaner than Baculum
and Bacularis.
And speaking of Baculum brings me to my main question, can bacula-web and
baculum/Bacularis exist on same server??
I think i need both. seeing as this is more of a reporting site, which
would be good for people that need to check on whether or not backups are
running and performing correctly(by the way, are there any functions for
alerts or warning on failed backup Jobs  ?) while Baculum provides a GUI on
creating jobs and backup tasks

Again thanks for all the work you put into this project

Sincerely /Mehrdad

**Comment by dfranco** [comment]
> Hi David So, i decided to start again with a fresh vm, one positive side of running VMs for this kinda work, and this time it worked, not sure if i had missed something before or what it was but i did two things differently First, after installing PHP and Apache, i made sure that php and apache worked, created a simple phpinfo() page and made it work(it did not at first, not sure why, could have been some mistake i made at installation but i had to add a few lines to the httpd's conf files) Second i just added the virtual site as mentioned in in the manuals and left the debian part of linking the files alone(in my previous installation i tried to do the equivalent of the debian a2en* and created the sites enable folders etc., this time, i did not) Also, i followed the installation steps more carefully and it worked . Whether it was for the two things i mentioned above or just following the instruction more closely, i don't know butit works I really like the esthetics and page design, it is way cleaner than Baculum and Bacularis. And speaking of Baculum brings me to my main question, can bacula-web and baculum/Bacularis exist on same server?? I think i need both. seeing as this is more of a reporting site, which would be good for people that need to check on whether or not backups are running and performing correctly(by the way, are there any functions for alerts or warning on failed backup Jobs ?) while Baculum provides a GUI on creating jobs and backup tasks Again thanks for all the work you put into this project Sincerely /Mehrdad
> […](#)
> On Wed, Apr 3, 2024 at 12:49 PM Davide Franco ***@***.***> wrote: Hi @Mehrdad-R10 <https://github.com/Mehrdad-R10> Thanks for the kind words about the project. Regarding the assets/protected error, please make sure you don't have SELinux enforced. Also, did you specify the basepath in config.php ? A dump of the web server config and application/config/config.php could be helpful to identify the problem. Best, — Reply to this email directly, view it on GitHub <[#221 (reply in thread)](https://github.com/bacula-web/bacula-web/discussions/221#discussioncomment-8995534)>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/BFWMITUJTISRVAHM6NUHNTLY3PNE3AVCNFSM6AAAAABFTJOLXWVHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM4DSOJVGUZTI> . You are receiving this because you were mentioned.Message ID: ***@***.***>

Hi @Mehrdad-R10 

First of all, thanks for your feedback, really appreciated, and I'm happy to know you manage to make it works on your vm :)

I'll seriously put some efforts on updating and reorganazing the documentation, which is too messy and incomplete in my opinion.

Let me try to give you some feedbacks / answers here

- can bacula-web and baculum/Bacularis exist on same server ? 
  I don't see see why it wouldn't be possible, if you face any issue please let me know, I'll help as much as I can
- are there any functions for alerts or warning on failed backup Jobs ?
  next major version will include major changes, I hope everyone will enjoy every single new feature
- Again thanks for all the work you put into this project 
  What I enjoy even more is when I receive such positive feedback from community users, but thanks for the kind words, appreciaed

Best regards
